### DISCUSSION TITLE
Why needs Kurtosis solution?

### DISCUSSION THREAD

**Discussion by ericobi** [kurtosis-tech/kurtosis#821] [Q&A] [answered]
Many projects have Docker-based solutions and sometimes shell scripts to run all containers in sequence.

What is the advantage of Kurtosis?

**Comment by ericobi** [comment, marked as answer]
Answer from @leeederek 

Kurtosis excels when it comes to complex systems and is basically a layer above Docker or Kubernetes that focuses specifically on the level of the entire environment. 

I can imagine that those Docker setups (or libraries or homemade CLI) work great for simple, smaller systems. However, as your system gets larger and more complex (i.e. more microservices & dependencies), then that is when those Docker setups and arbitrary bash scripts become difficult to maintain or use. From talking to our users, it gets painful in these cases because .yaml just doesn't cut it & also these setups are not very reproducible (e.g. works on your machine, but not on mine because of diff file paths, etc), and do not portable (i.e. bringing it into k8s or in CI requires different setup logic). 

One particular example I will call out is when a distributed system requires sequential setup logic & handling of dynamically generated dependencies - this use case is what Kurtosis was initially built to handle via Starlark & the Kurtosis orchestration engine.

**Comment by leeederek** [comment]
Thank you @ericobi for moving the conversation here :)
